lib/plugins/processor/post.js

```diff
@@ -121,7 +121,7 @@ module.exports = ctx => {
       }
 
       if (data.permalink) {
-        data.slug = data.permalink;
+        data.__permalink = data.permalink;
```

Comment:
BREAKING CHANGES.

`data.slug` won't be override by `data.permalink`. That means `data.slug` will still come from the original path relative to `source/_posts`.

Also,  a new `__permalink` is created to avoid overriding `Post.virtual('permalink').get.